src/correctionlib/highlevel.py

```diff
@@ -130,6 +131,49 @@ def _wrap_awkward(
     return awkward.transform(tocall, *array_args)
 
 
+def _call_dask_correction(
+    correction: Any,
+    *args: Union["numpy.ndarray[Any, Any]", str, int, float],
+):
+    return _wrap_awkward(correction._base.evalv, *args)
+
+
+def _wrap_dask_awkward(
+    correction: Any,
+    *args: Union["numpy.ndarray[Any, Any]", str, int, float],
+) -> Any:
+    import dask.delayed
+    import dask_awkward
+
+    if version.parse(dask_awkward.__version__) < _min_version_dak:
+        raise RuntimeError(
+            f"""imported dask_awkward is version {dask_awkward.__version__} < {str(_min_version_dak)}
+            This version of dask_awkward includes several useful bugfixes and functionality extensions.
+            Please upgrade dask_awkward.
+            """
+        )
+
+    if not hasattr(correction, "_delayed_correction"):
+        setattr(  # noqa: B010
+            correction,
+            "_delayed_correction",
+            dask.delayed(correction),
+        )
+
+    correction_meta = _wrap_awkward(
+        correction._base.evalv,
+        *(arg._meta if isinstance(arg, dask_awkward.Array) else arg for arg in args),
```

Comment:
Is there an abstract type for things that have `_meta`? If one were to pass a dask.array here what would happen?

Reply:
Though, there appears to be no base class that defines this spec, it is just a common pattern.

Reply:
Do I understand correctly that currently this supports dask-awkward arrays, but not dask arrays? And in the case of the latter, it emits a reasonable error message?

Reply:
Yep!